bundle/bundle-writer.go

```diff
@@ -0,0 +1,352 @@
+    Name     string   `json:"name"`
+    Selector string   `json:"selector"`
+    Tags     []string `json:"tags"`
+}
+
+
+func CreateSkycapFiles(outputDir string,
+						templateRepository string,
+						branch string,
+						pack_name string,
+						githubURL string,
+						services []*common.Service,
+						databases []common.Database) error {
+
+	if templateRepository == "" {
+		//no stencil template defined for this pack, print an error and do nothing
+		fmt.Printf("Sorry but there is no stencil template for this language/framework yet\n")
+		return nil
+	}
+	//Create .bundle directory structure if it doesn't exist
+	skycapFolder := filepath.Join(outputDir, "skycap")
+	err := createBundleFolderStructure(skycapFolder)
+	defer os.RemoveAll(filepath.Join(skycapFolder, "temp"))
+
+	//create manifest.json file and start filling
+	manifestFile, err := loadManifest()
+	if err != nil {
+		return err
+	}
+
+	manifestFile, err = saveEnvVars(pack_name, getEnvVars(services, databases), manifestFile, skycapFolder)
+	if err != nil {
+		return err
+	}
+
+	manifestFile, err = addDatabase(manifestFile, databases)
+
+	manifestFile, err = getRequiredStencils(
+		templateRepository,
+		branch,
+		outputDir,
+		services,
+		skycapFolder,
+		manifestFile,
+		githubURL)
+
+	if err != nil {
+		return err
+	}
+
+	manifestFile, err = addMetadatas(manifestFile)
+
+	if err != nil {
+		return err
+	}
+
+	err = saveManifest(skycapFolder, manifestFile)
+	if err !=nil {
+		return err
+	}
+	return err
+}
+
+// downloading templates from github and putting them into homedir
+func getStencilTemplateFile(templateRepository string, skycapFolder string, filename string, branch string) (string, error) {
+
+	//Download templates.json file
+	manifest_path := templateRepository + filename // don't need to use filepath since it's a URL
+	temp_folder := filepath.Join(skycapFolder, "temp")
+	down_err := common.DownloadSingleFile(temp_folder, common.DownloadFile{URL: manifest_path, Name: filename}, branch)
+	if down_err != nil {
+		return "", down_err
+	}
+	return filepath.Join(temp_folder, filename), nil
+}
+
+func getEnvVars (servs []*common.Service, databases []common.Database) (map[string]string) {
+	var envas = make(map[string]string)
+	for _, envVarArray := range servs {
+		for _, envs := range envVarArray.EnvVars {
+			envas[envs.Key] = envs.Value
+		}
+	}
+	return envas
+}
+
+func createBundleFolderStructure(baseFolder string) error {
+	var folders = [6]string{"stencils", "policies", "stencil-group", "helm-releases", "temp", "configurations"}
+	for _, subfolder := range folders {
+		folder := filepath.Join(baseFolder, subfolder)
+		err := os.MkdirAll(folder, 0777)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func getRequiredStencils(templateRepository string,
+						branch string,
+						outputDir string,
+						services []*common.Service,
+						skycapFolder string,
+						manifestFile ManifestBundle,
+						githubURL string) 	(ManifestBundle, error){
```

Comment:
return should be `(*ManifestBundler, error)` and you should be able to use `return nil, err` if there is an error.